xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/hibernate/query/HqlQueryExecutor.java

```diff
@@ -103,32 +112,39 @@ public void initialize() throws InitializationException
         configuration.addInputStream(Util.getResourceAsStream(MAPPING_PATH));
     }
 
-    private Set<String> getAllowedNamedQueries()
+    private Set<String> getSafeNamedQueries()
     {
-        if (this.allowedNamedQueries == null) {
+        if (this.safeNamedQueries == null) {
             synchronized (this) {
-                if (this.allowedNamedQueries == null) {
-                    this.allowedNamedQueries = new HashSet<>();
+                if (this.safeNamedQueries == null) {
+                    this.safeNamedQueries = new HashSet<>();
 
                     // Gather the list of allowed named queries
                     Collection<NamedQueryDefinition> namedQueries =
                         this.hibernate.getConfigurationMetadata().getNamedQueryDefinitions();
                     for (NamedQueryDefinition definition : namedQueries) {
-                        if (HqlQueryUtils.isSafe(definition.getQuery())) {
-                            this.allowedNamedQueries.add(definition.getName());
+                        try {
+                            if (isSafe(definition.getQuery())) {
+                                this.safeNamedQueries.add(definition.getName());
+                            }
+                        } catch (QueryException e) {
+                            this.logger.warn("Failed to validate named query [{}] with statement [{}]: {}",
+                                definition.getName(), definition.getQuery(), ExceptionUtils.getRootCauseMessage(e));
                         }
                     }
                 }
             }
         }
 
-        return this.allowedNamedQueries;
+        return this.safeNamedQueries;
     }
 
     /**
      * @param statementString the statement to evaluate
      * @return true if the select is allowed for user without PR
+     * @deprecated
      */
+    @Deprecated(since = "17.0.0RC1, 16.10.2, 15.10.16, 16.4.6")
```

Comment:
What's the motivation behind deprecating this method and introducing a new one with the changed behavior, in particular if the new behavior is exactly the same without configuration?

Reply:
Just the fact that this one is static, and I can't implement the new version in a static (because it needs to access components).